opentracing/tracer.py

```diff
@@ -40,31 +41,84 @@ class Tracer(object):
     def __init__(self):
         self._noop_span_context = SpanContext()
         self._noop_span = Span(tracer=self, context=self._noop_span_context)
+        self._active_span_source = NoopActiveSpanSource()
 
-    def start_span(self,
-                   operation_name=None,
-                   child_of=None,
-                   references=None,
-                   tags=None,
-                   start_time=None):
+    @property
+    def active_span_source(self):
+        return self._active_span_source
+
+    @property
+    def active_span(self):
+        return self._active_span_source.active_span
+
+    def start_active_span(self, operation_name, tags=None, start_time=None):
```

Comment:
this API (unnecessarily) couples two things:
1. whether the *current active span* should be consulted as an implicit parent, and
2. whether the *span we are starting* should be made active

`start_active_span` is supposed to be about the latter thing... IMO, both `start_active_span` *and* `start_manual_span` should consult the current active span and use it as a parent if no references were provided.

I.e., this function signature should have an optional `references=` param.